sonar-scala-plugin/src/test/java/org/sonarsource/scala/plugin/ScoverageSensorTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.scala.plugin;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.DefaultFileSystem;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.internal.DefaultSensorDescriptor;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.internal.MapSettings;
+import org.sonar.api.utils.log.LogTester;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ScoverageSensorTest {
+
+
+  private static final Path COVERAGE_DIR = Paths.get("src", "test", "resources", "coverage");
+  private static final String MODULE_KEY = "/Absolute/Path/To/";
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  private static ScoverageSensor scoverageSensor = new ScoverageSensor();
+
+  @Test
+  public void test_descriptor() {
+    DefaultSensorDescriptor sensorDescriptor = new DefaultSensorDescriptor();
+    scoverageSensor.describe(sensorDescriptor);
+    assertThat(sensorDescriptor.name()).isEqualTo("Scoverage sensor for Scala coverage");
+  }
+
+  @Test
+  public void testFilesCoverage() throws IOException {
+    Path baseDir = COVERAGE_DIR.toAbsolutePath();
+    Path reportPath = baseDir.resolve("scoverage.xml");
+
+    SensorContextTester context = getSensorContext(reportPath.toString(), "file1.scala", "file2.scala");
+
+    new ScoverageSensor().execute(context);
+
+    String fileKey1 = MODULE_KEY + ":file1.scala";
+    String fileKey2 = MODULE_KEY + ":file2.scala";
+
+    //File1
+    assertThat(context.lineHits(fileKey1, 6)).isNull();
+    assertThat(context.lineHits(fileKey1, 7)).isEqualTo(1);
+    assertThat(context.lineHits(fileKey1, 8)).isNull();
```

Comment:
only non executable lines are tested with: `.isNull()`
but executable lines, not covered, should be: `.isEqualTo(0)`